### DISCUSSION TITLE
settings_changed() callback behaviour

### DISCUSSION THREAD

**Discussion by dresco** [grblHAL/core#312] [General]
Just adding NVS support to my plugin, and a bit confused about the behaviour of the settings_changed() callback. 

This does get called when settings are first loaded, but not when my plugin settings are changed interactively through $xxx=yyy. Is that the expected behaviour?

My use case is that I want to copy something out of my settings structure into another variable. Was assuming settings_changed would be the way to do this..

Thanks!

**Comment by terjeio** [comment]
`hal.settings_changed` is called on load or when core settings are changed, there is a separated `on_changed` event that has to be subscribed to for plugin specific setting, see the [fans plugin](https://github.com/grblHAL/Plugin_fans/blob/4a8492609d5ec07ea9564919dcf444f6df3b53e8/fans.c#L439-L464) for an example.

`hal.settings_changed` should really be moved to `grbl.on_settings_changed`, but it is a breaking change so I have hesitated doing so...

**Reply by dresco** [reply]
Ah yes I see now. That's perfect, thanks!
